base/server/python/pki/server/deployment/__init__.py

```diff
@@ -1251,11 +1248,54 @@ def configure_ca(self, subsystem):
             if serial_transfer:
                 subsystem.set_config('dbs.serialCloneTransferNumber', serial_transfer)
 
-            if cert_id_generator == 'legacy2':
-                serial_dn = 'ou=certificateRepository,ou=ranges_v2'
-            else:
-                serial_dn = 'ou=certificateRepository,ou=ranges'
-            subsystem.set_config('dbs.serialRangeDN', serial_dn)
+            subsystem.set_config('dbs.serialRangeDN', 'ou=certificateRepository,ou=ranges')
+
+        elif cert_id_generator == 'legacy2':
+            subsystem.set_config('dbs.beginSerialNumber', '0x1')  # hex
+            subsystem.set_config('dbs.endSerialNumber', '0x10000000')  # hex
+            subsystem.set_config('dbs.serialIncrement', '0x10000000')  # hex
+            subsystem.set_config('dbs.serialLowWaterMark', '0x2000000')  # hex
+            subsystem.set_config('dbs.serialCloneTransferNumber', '0x10000')  # hex
+
+            if config.str2bool(self.mdict['pki_random_serial_numbers_enable']):
+                subsystem.set_config('dbs.enableRandomSerialNumbers', 'true')
+            subsystem.set_config('dbs.randomSerialNumberCounter', '0')
```

Comment:
I'm thinking to disallow this param for SSNv2 and ask the admin to use RSNv3 instead. We can do that separately later.